test/e2e/specs/editor/various/compatibility-classic-editor.spec.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Compatibility with classic editor', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test( 'Should not apply autop when rendering blocks', async ( {
+		page,
+		editor,
+	} ) => {
+		// Insert Html
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '/html' );
+		await expect(
+			page.locator( 'role=option[name="Custom HTML"i][selected]' )
+		).toBeVisible();
+
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '<a>' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Random Link' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '</a> ' );
+		// Publish Post
+		await editor.publishPost();
+
+		// View Post
+		await page.locator( 'a.components-button.is-primary' ).click();
+
+		// Check the content doesn't contain <p> tags.
+		await page.waitForSelector( '.entry-content' );
+		const content = await page.$eval( '.entry-content', ( element ) =>
+			element.innerHTML.trim()
+		);
+		expect( content ).toMatchSnapshot();
```

Comment:
It's a bit unfortunate that we don't have accessible selectors for `.entry-content` for now 🤔 . However, I think we can just assert the output as Playwright will implicitly wait for the element. Also, I think we don't need the snapshot.

```suggestion
		// No accessible selector for now.
		const content = page.locator( '.entry-content' );
		await expect.poll( () => content.innerHTML() ).toBe(`<a>
Random Link
</a>`);
```